ReSwift/CoreTypes/Store.swift

```diff
@@ -17,14 +17,26 @@ import Foundation
  */
 public class Store<State: StateType>: StoreType {
 
+    typealias SubscriptionType = Subscription<State>
+
     // TODO: Setter should not be public; need way for store enhancers to modify appState anyway
 
     /*private (set)*/ public var state: State! {
         didSet {
-            subscriptions.forEach {
+            var indexesToRemove: [Int] = []
+            for (index, subscription) in subscriptions.enumerate() {
+                guard let subscriber = subscription.subscriber else {
+                    indexesToRemove.append(index)
+                    return
```

Comment:
Ah yes, that makes more sense. There's indeed no need to implicitly unwrap
here! Optional chaining it is!

--
Benjamin Encz

Find me on twitter: @benjaminencz <https://twitter.com/benjaminencz>

On Thu, Feb 4, 2016 at 2:42 PM, Virgilio Favero Neto <
notifications@github.com> wrote:

> In ReSwift/CoreTypes/Store.swift
> <https://github.com/ReSwift/ReSwift/pull/70#discussion_r51950264>:
>
> >      // TODO: Setter should not be public; need way for store enhancers to modify appState anyway
> >
> >      /*private (set)*/ public var state: State! {
> >          didSet {
> > -            subscriptions.forEach {
> > +            var indexesToRemove: [Int] = []
> > +            for (index, subscription) in subscriptions.enumerate() {
> > +                guard let subscriber = subscription.subscriber else {
> > +                    indexesToRemove.append(index)
> > +                    return
>
> I'm happy with your suggestion.
>
> I'm leaning towards keeping the optional chaining when accessing the
> subscriber instead of implicit unwrapping it. There is no cost for the
> call, and it saves potential crashes caused by subscribers being
> deinitalized while in the forEach. Does that make sense?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReSwift/ReSwift/pull/70/files#r51950264>.
>